### PR TITLE
Performance Item 24, Clear buffers then null out request and response

### PR DESCRIPTION
**Pull request by uberskigeek**
Performance improvement.  Setting the HttpResponseImpl and HttpRequestImpl to null in HttpServiceContextImpl.clear after clearing their buffers

### CODE REVIEW

**Comment by uberskigeek** [comment]
#build

File: dev/com.ibm.ws.transport.http/src/com/ibm/ws/genericbnf/internal/BNFHeadersImpl.java [hunk 1]
```diff
@@ -1713,6 +1713,9 @@ private void clearAllHeaders() {
             elem.destroy();
             elem = next;
         }
+
+        //storage.clear();
+        //storage = new HashMap<Integer, HeaderElement>();
```
Comment (jhanders34) [inline dev/com.ibm.ws.transport.http/src/com/ibm/ws/genericbnf/internal/BNFHeadersImpl.java:1718]:
```suggestion
```
Please don't leave in commented out code.

File: dev/com.ibm.ws.transport.http/src/com/ibm/ws/http/channel/internal/HttpServiceContextImpl.java [hunk 2]
```diff
@@ -1044,31 +1044,23 @@ public void clear() {
...
         if (null != this.myRequest) {
-            if (this.bIsRequestOwner) {
-                this.myRequest.clear();
-            } else {
-                if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) {
-                    Tr.debug(tc, "Request not mine, skipping clear()");
-                }
-                this.myRequest = null;
-            }
+            this.myRequest.clearBuffers();
```
Comment (jhanders34) [inline dev/com.ibm.ws.transport.http/src/com/ibm/ws/http/channel/internal/HttpServiceContextImpl.java:1049]: Should we still check `if (this.bIsRequestOwner)` ?

File: dev/com.ibm.ws.transport.http/src/com/ibm/ws/http/channel/internal/HttpServiceContextImpl.java [hunk 3]
```diff
@@ -1044,31 +1044,23 @@ public void clear() {
 
         // clear the request message if we're the "owner" otherwise clear
         // our reference to it
+
         if (null != this.myRequest) {
-            if (this.bIsRequestOwner) {
-                this.myRequest.clear();
```
Comment (jhanders34) [inline dev/com.ibm.ws.transport.http/src/com/ibm/ws/http/channel/internal/HttpServiceContextImpl.java:1049]: If clear() isn't going to be called any longer should it just be removed?  Also as discussed if we aren't going to use the pool any longer, that can also be removed.

File: dev/com.ibm.ws.transport.http/src/com/ibm/ws/genericbnf/internal/BNFHeadersImpl.java (hunk 1 above)
Comment (uberskigeek) [reply dev/com.ibm.ws.transport.http/src/com/ibm/ws/genericbnf/internal/BNFHeadersImpl.java:1718]: Duh....

**Comment by uberskigeek** [comment]
This is superseded by PR  #23584

**Pull request closed without merging**